mcts_new.py

```diff
@@ -0,0 +1,280 @@
+from __future__ import annotations
+
+import dataclasses
+import enum
+
+import einops
+import numpy as np
+import torch
+
+import skyjo as sj
+import skynet
+
+
+# MARK: Enums
+class MCTSSelectionMethod(enum.Enum):
+    UCB = "UCB"  # AlphaZero UCB
```

Comment:
I feel like subclassing might be more ergonomic/easier to separate into individual files/notebooks here